### PR TITLE
MESOS: Extract slave hostname registry code in its own module

### PR DESCRIPTION
**Pull request by sttts**
- remove bleeding of registry-internal objects, without any locking
- rename from SlaveStorage to Registry which fits much better to what
  it actually does

### CODE REVIEW

**Comment by sttts** [comment]
@jdef for internal review

**Comment by jdef** [comment]
lgtm

**Comment by alex-mohr** [comment]
Shippable failed, so kicking it once.
"NodeController
Successful get nodes 127.0.0.1 {{.spec.unschedulable}}: 
"127.0.0.1" patched
Successful get nodes 127.0.0.1 {{.spec.unschedulable}}: true
Error from server: node "127.0.0.1" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
!!! Error in ./hack/test-cmd.sh:776
  'kubectl patch "${kube_flags[@]}" nodes "127.0.0.1" -p='{"spec":{"unschedulable":null}}'' exited with status 1
Call stack:
  1: ./hack/test-cmd.sh:776 runTests(...)
  2: ./hack/test-cmd.sh:822 main(...)
Exiting with status 1"

**Comment by alex-mohr** [comment]
@k8s-bot test this please (no test results found)

File: contrib/mesos/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -423,7 +376,7 @@ func (k *KubernetesScheduler) StatusUpdate(driver bindings.SchedulerDriver, task
 			} // else, we don't really care about FINISHED tasks that aren't registered
 			return
 		}
-		if _, exists := k.slaves.getSlave(taskStatus.GetSlaveId().GetValue()); !exists {
+		if hostName := k.slaveHostNames.HostName(taskStatus.GetSlaveId().GetValue()); hostName != "" {
```
Comment (jdef) [inline contrib/mesos/pkg/scheduler/scheduler.go]: this conditional looks wrong, see https://github.com/mesosphere/kubernetes-mesos/issues/522

Comment (sttts) [inline contrib/mesos/pkg/scheduler/scheduler.go]: Good spot.

**Pull request merged**